packages/signal-polyfill/src/wrapper.ts

```diff
@@ -234,10 +234,16 @@ export namespace subtle {
       this.#assertSignals(signals);
 
       const node = this[NODE];
+      assertConsumerNode(node);
+
       node.dirty = false;  // Give the watcher a chance to trigger again
       const prev = setActiveConsumer(node);
+
+      const producerNodeSet = new Set(node.producerNode);
```

Comment:
Seems a bit expensive to reconstruct this set on each watch call. Is there any way to reduce this cost, whether by maintaining the set across several method calls, or somehow avoiding constructing a set and using the producerNode data structure in some other way instead?

Reply:
> ... are much faster collections as compared to sets / maps.
Makes sense. Does that generally include traversing an array though ? If so we can just filter out duplicates first instead of constructing a set. What do you think ?

Reply:
Something like `signals.filter(s => node.producerNode.includes(s)).forEach(producerAccessed)` should work ( or even just a for loop without creating a new array with the filter ). We are at O(n.m) though, since we "traverse" producerNode for each signal.

Reply:
Updated in [0684e8a](https://github.com/tc39/proposal-signals/pull/191/commits/0684e8a5315cd781822bff64a4cdbe38cbfdaf22) to use `.includes` instead. It's not the ideal option perf wise though. If we want better performance i think we should rethink the `producerNode` data structure all together. @littledan @pkozlowski-opensource what do you think ?